### PR TITLE
ST: system test resources refactor

### PR DESCRIPTION
**Pull request by Frawless**
### Type of change

- Refactoring

### Description

This PR contains refactor of system test resources:
- each cluster component has its own resource class 
- method and class resources are stored in `ResourceManager`
- resources could be created from everywhere now (manager is singleton) 
- removed some lines from log whitelist (reason for the logs fixed) 

### Checklist

- [x] Make sure all tests pass

### CODE REVIEW

**Comment by Frawless** [comment]
@strimzi-ci run tests

**Comment by Seequick1** [comment]
@strimzi-ci run tests test_only

**Comment by Frawless** [comment]
@strimzi-ci run tests test_only

**Comment by Frawless** [comment]
@strimzi-ci run tests test_only testcase=KafkaST

**Comment by Frawless** [comment]
@strimzi-ci run tests test_only testcase=KafkaST

**Comment by Frawless** [comment]
@strimzi-ci run tests test_only testcase=KafkaST

**Comment by Frawless** [comment]
@strimzi-ci run tests test_only testcase=KafkaST

**Comment by Frawless** [comment]
@strimzi-ci run tests test_only testcase=KafkaST

**Comment by Frawless** [comment]
@strimzi-ci run tests test_only

**Comment by Frawless** [comment]
I will do rebase when https://github.com/strimzi/strimzi-kafka-operator/pull/2244 will be merged to avoid doing rebase multiple times. i think we can wait with review after that.

**Comment by Frawless** [comment]
@strimzi-ci run tests

File: systemtest/src/main/java/io/strimzi/systemtest/utils/StUtils.java [hunk 1]
```diff
@@ -651,6 +679,11 @@ public static void waitForKafkaUserDeletion(String userName) {
         LOGGER.info("Kafka user {} deleted", userName);
     }
 
+    public static void waitForKafkaUserIncreaseObserverGeneration(long observation, String userName) {
+        TestUtils.waitFor("Wait until increase observation level from " + observation + " for user " + userName, Constants.GLOBAL_POLL_INTERVAL, Constants.TIMEOUT_FOR_SECRET_CREATION,
+            () -> observation < KafkaUserResource.kafkaUserClient().inNamespace(kubeClient().getNamespace()).withName(userName).get().getStatus().getObservedGeneration());
```
Comment (tombentley) [inline systemtest/src/main/java/io/strimzi/systemtest/utils/StUtils.java]: Very long line.

File: systemtest/src/main/java/io/strimzi/systemtest/utils/StUtils.java [hunk 2]
```diff
@@ -643,6 +644,33 @@ public static void waitForSecretReady(String secretName) {
...
+                }
+            });
+        LOGGER.info("Kafka cluster {} secrets deleted", clusterName);
+    }
+
+    public static void waitForKafkaUserCreation(String userName) {
+        LOGGER.info("Waiting for Kafka user creation {}", userName);
+        waitForSecretReady(userName);
+        TestUtils.waitFor("Waits for Kafka user deletion " + userName, Constants.POLL_INTERVAL_FOR_RESOURCE_READINESS, Constants.TIMEOUT_FOR_RESOURCE_READINESS,
+            () -> KafkaUserResource.kafkaUserClient().inNamespace(kubeClient().getNamespace()).withName(userName).get().getStatus().getConditions().get(0).getType().equals("Ready")
```
Comment (tombentley) [inline systemtest/src/main/java/io/strimzi/systemtest/utils/StUtils.java]: Very long line.

File: systemtest/src/main/java/resources/ResourceManager.java [hunk 3]
```diff
@@ -0,0 +1,308 @@
...
+        pointerResources = methodResources;
+    }
+
+    public static void setClassResources() {
+        LOGGER.info("Setting pointer to class resources");
+        pointerResources = classResources;
+    }
+
+    @SuppressWarnings("unchecked")
+    public static <T extends HasMetadata> T deleteLater(MixedOperation<T, ?, ?, ?> x, T resource) {
```
Comment (tombentley) [inline systemtest/src/main/java/resources/ResourceManager.java]: Can we rename `x` → `operation`

File: systemtest/src/main/java/resources/crd/KafkaMirrorMakerResource.java [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+    public static MixedOperation<KafkaMirrorMaker, KafkaMirrorMakerList, DoneableKafkaMirrorMaker, Resource<KafkaMirrorMaker, DoneableKafkaMirrorMaker>> kafkaMirrorMakerClient() {
+        return Crds.mirrorMakerOperation(ResourceManager.kubeClient().getClient());
+    }
+
+    public static DoneableKafkaMirrorMaker kafkaMirrorMaker(String name, String sourceBootstrapServer, String targetBootstrapServer, String groupId, int mirrorMakerReplicas, boolean tlsListener) {
+        KafkaMirrorMaker kafkaMirrorMaker = getKafkaMirrorMakerFromYaml(PATH_TO_KAFKA_MIRROR_MAKER_CONFIG);
+        return deployKafkaMirrorMaker(defaultKafkaMirrorMaker(kafkaMirrorMaker, name, sourceBootstrapServer, targetBootstrapServer, groupId, mirrorMakerReplicas, tlsListener).build());
+    }
+
+    private static KafkaMirrorMakerBuilder defaultKafkaMirrorMaker(KafkaMirrorMaker kafkaMirrorMaker, String name, String sourceBootstrapServer, String targetBootstrapServer, String groupId, int kafkaMirrorMakerReplicas, boolean tlsListener) {
```
Comment (tombentley) [inline systemtest/src/main/java/resources/crd/KafkaMirrorMakerResource.java]: Long line.

File: systemtest/src/main/java/io/strimzi/systemtest/utils/StUtils.java [hunk 5]
```diff
@@ -643,14 +644,52 @@ public static void waitForSecretReady(String secretName) {
...
+                    return false;
+                }
+            });
+        LOGGER.info("Kafka cluster {} secrets deleted", clusterName);
+    }
+
+    public static void waitForKafkaUserCreation(String userName) {
+        LOGGER.info("Waiting for Kafka user creation {}", userName);
+        waitForSecretReady(userName);
+        TestUtils.waitFor("Waits for Kafka user deletion " + userName,
```
Comment (ppatierno) [inline systemtest/src/main/java/io/strimzi/systemtest/utils/StUtils.java]: this is about creation not deletion

File: systemtest/src/main/java/io/strimzi/systemtest/utils/StUtils.java [hunk 6]
```diff
@@ -643,14 +644,52 @@ public static void waitForSecretReady(String secretName) {
...
-        TestUtils.waitFor("Waits for Kafka user deletion " + userName, Constants.POLL_INTERVAL_FOR_RESOURCE_READINESS, Constants.TIMEOUT_FOR_RESOURCE_READINESS,
+        TestUtils.waitFor("Waits for Kafka user deletion " + userName,
+                Constants.POLL_INTERVAL_FOR_RESOURCE_READINESS, Constants.TIMEOUT_FOR_RESOURCE_READINESS,
             () -> Crds.kafkaUserOperation(kubeClient().getClient()).inNamespace(kubeClient().getNamespace()).withName(userName).get() == null
         );
         LOGGER.info("Kafka user {} deleted", userName);
     }
 
+    public static void waitForKafkaUserIncreaseObserverGeneration(long observation, String userName) {
+        TestUtils.waitFor("Wait until increase observation level from " + observation + " for user " + userName,
```
Comment (ppatierno) [inline systemtest/src/main/java/io/strimzi/systemtest/utils/StUtils.java]: I would use "observed generation" instead of "observation level"

**Pull request merged**